wire-ios-sync-engine/Source/Use cases/EnableAnalyticsSharingUseCase.swift

```diff
@@ -0,0 +1,75 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Countly
+import WireAnalytics
+
+// MARK: - EnableAnalyticsSharingUseCaseProtocol
+
+// sourcery: AutoMockable
+/// Protocol defining the interface for enabling analytics sharing.
+public protocol EnableAnalyticsSharingUseCaseProtocol {
+
+    /// Invokes the use case to enable analytics sharing.
+    func invoke()
+}
+
+// MARK: - EnableAnalyticsSharingUseCase
+
+/// Concrete implementation of the EnableAnalyticsSharingUseCaseProtocol.
+/// This struct is responsible for enabling analytics sharing for a specific user profile.
+public struct EnableAnalyticsSharingUseCase: EnableAnalyticsSharingUseCaseProtocol {
+
+    // MARK: - Properties
+
+    /// The analytics manager responsible for handling tracking operations.
+    private let analyticsManager: AnalyticsManagerProtocol?
```

Comment:
Why is this optional?

Reply:
I don't think it needs to be. Only in the case where we have analytics disabled for specific builds... 🤔 Not sure. Removed optionality here: https://github.com/wireapp/wire-ios/commit/d239e12ad69ed903c7dc4527af5b2b1d4d00fc61